### PR TITLE
Unique key with testcases data

### PR DESCRIPTION
**Pull request by rupaliT123**
Unique key with testcases data

### CODE REVIEW

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Please provide lint count and coverage as per merge request process already shared.

dev 
coverage:85%
 lint:0
 
UniqueKeyWithTestcasesData:
 coverage:85%
 lint:0

File: src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java [hunk 1]
```diff
@@ -368,8 +391,14 @@ public FormDataResponse updateFormData(FormDataSchema formDataSchema) throws Jso
         formDataDefinition.setCreatedOn(String.valueOf(document.get(CREATED_ON)));
         formDataDefinition.setUpdatedById(String.valueOf(loggedInUserId));
         formDataDefinition.setUpdatedOn(String.valueOf(Instant.now()));
-        mongoTemplate.getCollection(TP_RUNTIME_FORM_DATA +formId).updateOne(filter,Updates.combine(
-                Updates.set(FORM_DATA,formData),Updates.set(VERSION,version),Updates.set(FORM_META_DATA,formMetaData)));
+        try {
+            mongoTemplate.getCollection(TP_RUNTIME_FORM_DATA + formId).updateOne(filter, Updates.combine(
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java:395]: unnecessary change, please take care not to add them to commit

File: src/main/resources/errorMessages_en.properties [hunk 2]
```diff
@@ -35,4 +35,5 @@ AWGMENT-RUNTIME-FORM-1035=Filter should be given along with sort-by sort-order a
 AWGMENT-RUNTIME-FORM-1036=Unable to delete formData record with given formDataId and formId in elastic db
 AWGMENT-RUNTIME-FORM-1037=Regex condition failed for the given field: {0}
 AWGMENT-RUNTIME-FORM-1039=Please remove id from your payload since this is a new collection id should not be given
+AWGMENT-RUNTIME-FORM-1041=Provided already existing officialEmailId
```
Comment (VaibhavTechsophy) [inline src/main/resources/errorMessages_en.properties]: This is not a generic message please add a generic message.

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java [hunk 3]
```diff
@@ -202,6 +205,197 @@ void saveFormDataWithoutCollectionTest() throws IOException
...
+        formDataMap.put(UPDATED_ON,TEST_UPDATED_ON);
+        formDataMap.put(UPDATED_BY_ID,UPDATED_BY_USER_ID);
+        formDataMap.put(UPDATED_BY_NAME,UPDATED_BY_USER_NAME);
+        Mockito.when(mockMongoTemplate.save(any(),anyString())).thenThrow(new MongoException("Duplicate key"));
+        Mockito.when(mockObjectMapper.convertValue(any(),eq(FormDataDefinition.class))).thenReturn(new FormDataDefinition());
+        Assertions.assertThrows(InvalidInputException.class,()->mockFormDataServiceImpl.saveFormData(formDataSchemaTest));
+    }
+
+    @Test
+    void saveFormDataWithUniqueKeyCheck1() throws IOException
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java]: provide a name that explains hte test case

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Please fix generic message and others

File: src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java (hunk 1 above)
Comment (rupaliT123) [reply src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java:395]: It is important to catch an exceptions that may arise when updating the officialEmail field, particularly in cases where someone may have inadvertently entered a duplicate value . In these cases we have to handle this situation by catching exception

**Comment by rupaliT123** [comment]
dev 
coverage:85%
 lint:0
 
UniqueKeyWithTestCasesData:
 coverage:85%
 lint:0

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java (hunk 3 above)
Comment (rupaliT123) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java]: I have changed the name

File: src/main/resources/errorMessages_en.properties (hunk 2 above)
Comment (rupaliT123) [reply src/main/resources/errorMessages_en.properties]: "Duplicate key has been provided  " is the new generic constant

File: src/main/resources/errorMessages_en.properties [hunk 4]
```diff
@@ -35,4 +35,5 @@ AWGMENT-RUNTIME-FORM-1035=Filter should be given along with sort-by sort-order a
 AWGMENT-RUNTIME-FORM-1036=Unable to delete formData record with given formDataId and formId in elastic db
 AWGMENT-RUNTIME-FORM-1037=Regex condition failed for the given field: {0}
 AWGMENT-RUNTIME-FORM-1039=Please remove id from your payload since this is a new collection id should not be given
+AWGMENT-RUNTIME-FORM-1041=Duplicate key has been provided
```
Comment (VaibhavTechsophy) [inline src/main/resources/errorMessages_en.properties]: Please use the text-
"A duplicate value has been provided for a field marked as unique"

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Minor change on the error message

File: src/main/resources/errorMessages_en.properties (hunk 4 above)
Comment (rupaliT123) [reply src/main/resources/errorMessages_en.properties]: I have provided the same message as you have given

**Comment by rupaliT123** [comment]
dev
coverage:85%
lint:0

UniqueKeyWithTestCasesData:
coverage:85%
lint:0

**Review by VaibhavTechsophy — APPROVED** [review]
